### DISCUSSION TITLE
loop api naming

### DISCUSSION THREAD

**Discussion by awaelchli** [Lightning-AI/pytorch-lightning#8027] [General]
@Borda suggested the following for the API naming in [new loops](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/pytorch_lightning/loops/base.py):

`Loop.is_done` instead of `Loop.done`
`Loop.should_skip` instead of `Loop.skip`

Note, these are properties, not methods.

Do you prefer the longer name (less ambiguous) or shorter name (easier to remember)

@justusschock @ananthsub @tchaton @Borda 

Vote:
- short names 🚀 
- long names 🎉

**Comment by Borda** [comment]
> Note, these are properties, not methods.

that is true, so maybe the `()` make the job of separation action from the state...
probably keep just `skip` but would prefer the `is_` when you ask about state and return bool

**Comment by tchaton** [comment]
IMO, I prefer the current approach for the `done` property as this is going to be used in a negative form.

```
while not loop.is_done
```

`while not is done` doesn't sound right compared to `while not done`.

For skip, I think `Loop.should_skip`seems better.

Best,
T.C
